### ISSUE TITLE
Question about screen transition.

### ISSUE THREAD

**Issue by BruceLiCong** [SnoringCatGames/scaffolder#5]
https://github.com/SnoringCatGames/scaffolder/blob/ef78541f5ad58fa9e3efafe21124b475a1c01c2b/src/utils/screen_navigator.gd#L287
Is it : `is_open and !next_screen_was_already_shown` ?

From the previous logic `next_screen_was_already_shown` won't be true if `is_open` is false, so `is_open or next_screen_was_already_shown` exits redundant judgments.

And from my understanding, `is_open and !next_screen_was_already_shown` makes more sense for `is_forward`
.

**Comment by levilindsey** [comment]
Good question! I had some trouble remembering / figuring this one out...

tl;dr: Yes!

- This maintains a stack of "open" screens, so that the player can usually move "back" to the previous screen.
- In general, I intend for new screens to animate from a position on the right side of the screen in toward the center of the screen.
  - When going "back", we then animate the old screen from the center to off toward the right side.
  - The "game" screen is a special case. I always want it to be underneath other screens (lower z-index), so when navigating to the game screen as the next screen, I animate the previous screen off to the left.
    - So there is still the impression that we are moving from the left to the right when going "forward".
- I designed this system around the assumption that you'll never want the same screen type to appear twice on the "open" screen stack.
  - If we try to open a screen that was already on the stack, we instead "close" all of the later screens on the stack.
  - And this is why `next_screen_was_already_shown` is meant to be important.

All of this screen-handling logic is pretty arbitrary, and just the way I thought it would make sense for most of my games.

I can push a fix for this soon.

Sorry about the delay! December's a busy month for me. :) I'm very happy to keep answering any of your questions!

**Comment by levilindsey** [comment]
This should be fixed in [54ea911](https://github.com/SnoringCatGames/scaffolder/commit/54ea91178b37a281111b9ed9c67aa882cbee1db1)

**Comment by BruceLiCong** [comment]
Thanks very much for the confirmation and details sharing. Keep on studying the your awesome framework and 2d platformer navigation solution!
